pkg/blockstorage/azure/client.go

```diff
@@ -20,27 +20,37 @@ package azure
 
 import (
 	"context"
-
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
-	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/azure"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
 	"github.com/kanisterio/kanister/pkg/blockstorage"
 	"github.com/kanisterio/kanister/pkg/log"
 	"github.com/pkg/errors"
 )
 
 // Client is a wrapper for Client client
 type Client struct {
-	SubscriptionID  string
-	ResourceGroup   string
-	BaseURI         string
-	Authorizer      *autorest.BearerAuthorizer
-	DisksClient     *compute.DisksClient
-	SnapshotsClient *compute.SnapshotsClient
+	Cred           *azidentity.DefaultAzureCredential
+	SubscriptionID string
+	ResourceGroup  string
+	BaseURI        string
+	//https://github.com/Azure-Samples/azure-sdk-for-go-samples/blob/main/sdk/resourcemanager/compute/disk/main.go
+	//https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/arm/compute#DisksClient
+	DisksClient *armcompute.DisksClient
+	//https://github.com/Azure-Samples/azure-sdk-for-go-samples/blob/main/sdk/resourcemanager/compute/snapshot/main.go
+	//https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/arm/compute#SnapshotsClient
+	SnapshotsClient     *armcompute.SnapshotsClient
+	SKUsClient          *armcompute.ResourceSKUsClient
+	SubscriptionsClient *armsubscriptions.Client
 }
 
 // NewClient returns a Client struct
```

Comment:
[nit] comment should probably go to line 53 to match with function

Reply:
Done